DigitalLearningSolutions.Data/Services/GroupsService.cs

```diff
@@ -400,6 +429,53 @@ int centreId
             transaction.Complete();
         }
 
+        public void GenerateGroupsFromRegistrationField(GroupGenerationDetails groupDetails)
+        {
+            var isJobGroup = groupDetails.RegistrationField.Equals(RegistrationField.JobGroup);
+            var linkedToField = groupDetails.RegistrationField.LinkedToField;
+
+            (List<(int id, string name)> newGroupNames, string groupNamePrefix) = isJobGroup
+                ? GetJobGroupsAndPrefix()
+                : GetCustomPromptsAndPrefix(groupDetails.CentreId, groupDetails.RegistrationField.Id);

```

Comment:
Ideally using the enumeration class would allow us to ignore the value of `RegistrationField.Id` a bit more, and just pass `RegistrationField` in places like this. If you replace `CustomPromptNumber` in `CustomPrompt` with a `RegistrationField` that should improve things a bit.

Reply:
Reverted the changes, with redo them in a separate MR